AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/physx_collider.py

```diff
@@ -0,0 +1,318 @@
+            BASE = 'Shape Configuration|Cylinder'
+            RADIUS = 'Shape Configuration|Cylinder|Radius'
+            HEIGHT = 'Shape Configuration|Cylinder|Height'
+            SUBDIVISION = 'Shape Configuration|Cylinder|Subdivision'
+        class PhysicsAsset:
+            BASE = 'Shape Configuration|Asset'
+            PHYSX_MESH = 'Shape Configuration|Asset|PhysX Mesh'
+            ASSET_SCALE = 'Shape Configuration|Asset|Configuration|Asset Scale'
+            PHYSICS_MATERIALS_FROM_ASSET = 'Shape Configuration|Asset|Configuration|Physics materials from asset'
+
+        COLLISION_LAYER = 'Collider Configuration|Collision Layer'
+        COLLIDES_WITH = 'Collider Configuration|Collides With'
+        TRIGGER = 'Collider Configuration|Trigger'
+        SIMULATED = 'Collider Configuration|Simulated'
+        IN_SCENE_QUERIES = 'Collider Configuration|In Scene Queries'
+        OFFSET = 'Collider Configuration|Offset'
+        ROTATION = 'Collider Configuration|Rotation'
+        PHYSX_MATERIAL_ASSET = ']|'  # GH-12503 Needs a better property path
+        TAG = 'Collider Configuration|Tag'
+        RESET_OFFSET = 'Collider Configuration|Rest offset'
+        CONTACT_OFFSET = 'Collider Configuration|Contact offset'
+        DRAW_COLLIDER = 'Debug draw settings|Draw collider'
+        SHAPE = 'Shape Configuration|Shape'
+
+    # General Properties
+    #TODO: GHI #12632 - Figure out how to set dropdowns for Collison Layer and Collides With
+
+    def toggle_trigger(self) -> None:
+        """
+        Property Type, Default Visibility - ('bool', 'Visible')
+
+        Used to toggle the Physx Collider's Trigger Property.
+        """
+        self.component.toggle_property_switch(self.Path.TRIGGER)
+
+    def toggle_simulated(self) -> None:
+        """
+        Property Type, Default Visibility - ('bool', 'Visible')
+
+        Used to toggle the Physx Collider's Simulated Property.
+        """
+        self.component.toggle_property_switch(self.Path.SIMULATED)
+
+    def toggle_in_scene_queries(self) -> None:
+        """
+        Property Type, Default Visibility - ('bool', 'Visible')
+
+        Used to toggle the Physx Collider's In Scene Queries Property.
+        """
+        self.component.toggle_property_switch(self.Path.IN_SCENE_QUERIES)
+
+    def set_offset(self, x: float, y: float, z: float) -> None:
+        """
+        Property Type, Default Visibility - ('Vector3', 'Visible')
+
+        Used to set the Physx Collider's Offset Property.
+        """
+        self.component.set_component_property_value(self.Path.OFFSET, math.Vector3(float(x), float(y), float(z)))
+
+    def set_rotation(self, x: float, y: float, z: float) -> None:
+        """
+        Property Type, Default Visibility - ('Quaternion', 'Visible')
+
+        Used to set the Physx Collider's Rotation Property.
+        """
+        rotation = math.Quaternion()
+        rotation.SetFromEulerDegrees(math.Vector3(float(x), float(y), float(z)))
```

Comment:
x y and z are already specified as float in the parameters, could this just be `math.Vector3(x, y ,z))`

Reply:
Unfortunately, Python does not enforce data types passed in. Adding the data type in the function definition a form of "hinting" and helps with autocomplete in IDEs and informing a user of the function of the intended data types to be passed in.

Reply:
Removed this in favor of improving error messaging and not allowing a non-float from even getting to this point of the code.